packages/dds/tree/src/simple-tree/api/identifierIndex.ts

```diff
@@ -0,0 +1,270 @@
+					if (schemus !== undefined) {
+						const keyLocation = indexer(schemus);
+						if (keyLocation !== undefined) {
+							return makeGenericKeyFinder<TKey>(brand(keyLocation), isKeyValid);
+						}
+					} else {
+						fail("node is out of schema");
+					}
+				}
+			: (schemaIdentifier: TreeNodeSchemaIdentifier) => {
+					const schemus = indexableSchemaMap.get(schemaIdentifier);
+					if (schemus !== undefined) {
+						const keyLocation = indexer(schemus);
+						if (keyLocation !== undefined) {
+							return makeGenericKeyFinder<TKey>(brand(keyLocation), isKeyValid);
+						}
+					}
+				};
+
+	const index = new AnchorTreeIndex<TKey, TValue>(
+		(view as SchematizingSimpleTreeView<TFieldSchema>).getView().checkout.forest,
+		schemaIndexer,
+		(anchorNodes) => {
+			const simpleTreeNodes: TreeNode[] = [];
+			for (const a of anchorNodes) {
+				const simpleTree = getOrCreateSimpleTree(view, a);
+				if (!isTreeValue(simpleTree)) {
+					simpleTreeNodes.push(simpleTree);
+				}
+			}
+
+			if (hasElement(simpleTreeNodes)) {
+				return getValue(simpleTreeNodes);
+			}
+		},
+		(anchorNode: AnchorNode) => {
+			const simpleTree = getOrCreateSimpleTree(view, anchorNode);
+			if (!isTreeValue(simpleTree)) {
+				return treeNodeApi.status(simpleTree);
+			}
+		},
+	);
+
+	// all the type checking guarantees that we put nodes of the correct type in the index
+	// but it's not captured in the type system
+	return index as SimpleTreeIndex<TKey, TValue>;
+}
+
+/**
+ * An index that returns tree nodes given their associated identifiers.
+ *
+ * @alpha
+ */
+export type IdentifierIndex = SimpleTreeIndex<string, TreeNode>;
+
+function isStringKey(key: TreeIndexKey): key is string {
+	return typeof key === "string";
+}
+
+/**
+ * Creates an {@link IdentifierIndex} for a given {@link TreeView}.
+ *
+ * @alpha
+ */
+export function createIdentifierIndex<TSchema extends ImplicitFieldSchema>(
+	view: TreeView<TSchema>,
+): IdentifierIndex {
+	// For each node schema, find which field key the identifier field is under.
+	// This can be done easily because identifiers are their own field kinds.
+	const identifierFields = new Map<string, string>();
+	walkFieldSchema(view.schema, {
+		node: (schemus) => {
+			if (schemus instanceof ObjectNodeSchema) {
+				for (const fieldKey of schemus.fields.keys()) {
+					identifierFields.set(schemus.identifier, fieldKey);
+					break;
+				}
+			}
+		},
+	});
+
+	return createSimpleTreeIndex(
+		view,
+		(schemus) => identifierFields.get(schemus.identifier),
+		(nodes) => {
+			if (nodes.length > 1) {
+				throw new UsageError(
+					"cannot retrieve node from index: there are multiple nodes with the same identifier",
+				);
+			}
+
+			return nodes[0];
+		},
+		isStringKey,
+	);
+}
+
+/**
+ * Gets a simple tree from an anchor node
+ */
+function getOrCreateSimpleTree<TSchema extends ImplicitFieldSchema>(
+	view: TreeView<TSchema>,
+	anchorNode: AnchorNode,
```

Comment:
AnchorNodes are recounted objects. When passing them around the docs must be clear about lifetimes (ex: document that the caller must ensures it stays alive until the method returns, meaning that if the callee wants to hold onto it has to add a ref count.

I think you arn't actually using any anchorNode specific APis here, so maybe taking in an upPath (and noting its invalid for the calee to hold onto) might be simpler.